include/simdjson/numberparsing.h

```diff
@@ -562,9 +572,33 @@ static really_inline bool parse_number(const uint8_t *const buf, ParsedJson &pj,
       // we start anew, going slowly!!!
       return parse_float(buf, pj, offset, found_minus);
     }
-    double factor = power_of_ten[power_index];
-    factor = negative ? -factor : factor;
-    double d = i * factor;
+    double d = 0;
+    if (i != 0) {
+      components c = power_of_ten_components[power_index];
+      uint64_t factor_mantissa = c.mantissa;
+      int lz = leading_zeroes(i);
+      i <<= lz;
+      __uint128_t large_mantissa = (__uint128_t)i * factor_mantissa;
+      uint64_t upper = large_mantissa >> 64;
+      if (likely((upper & 0x1FF) != 0x1FF)) {
+        uint64_t mantissa = 0;
+        if (upper & (1ULL << 63)) {
+          mantissa = upper >> 10;
+        } else {
+          mantissa = upper >> 9;
+          lz++;
+        }
+        mantissa += mantissa & 1;
+        mantissa >>= 1;
+        mantissa &= ~(1ULL << 52);
+        uint64_t real_exponent = c.exp + 1023 + (127 - lz);
+        mantissa |= real_exponent << 52;
+        mantissa |= ((uint64_t)negative) << 63;
+        d = *((double *)&mantissa);
```

Comment:
This is almost certainly not 'safe'. I recommend doing... `memcpy(&d,&mantissa,8)`.